wave_lang/kernel/ops/wave_ops.py

```diff
@@ -40,6 +40,41 @@
 PlaceholderT = TypeVar("PlaceholderT", bound="Placeholder")
 
 
+def read_meets_hw_transpose_requirements(
+    read: Read, constraints: list[Constraint]
+) -> bool:
+    from ..wave.utils.general_utils import find_index_bounds
+
+    if read.bounds is not None:
+        return False
+
+    bounds = find_index_bounds(
+        constraints, read.index, read.vector_shapes, read.type.symbolic_shape
+    )
+
+    if bounds is not None:
+        return False
+
+    if read.has_identity_mapping():
+        return False
+
+    if len(list(read.index.keys())) != 2:
```

Comment:
This check looks incomplete, should we check mapping corresponds to the transpose?

Reply:
Now I'm thinking separate op or ReadOp flag op weren't the bad ideas :). Anyway, lets merge this in current form to unblock things and see what we can improve later.